Eliminate burner accounts (#4673) (#4854)
DataBiosphere/azul: README.md

The Terra ecosystem is tightly integrated with Google's authentication
infrastructure, and the same two types of accounts mentioned in the previous
section are used to authenticate against SAM and [Terra Data Repository]
(TDR).

> Meaning that there are now at least two Google accounts at play:

Reviewer: Join with previous paragraph. Can't start a sentence like that with a gerund.

1) your individual Google account ("your account"),

2) a service account for each shared or personal Azul deployment.

You use your account to interact with Google Cloud in general, along with both

> production and non-production instances of Terra, SAM, and TDR, assuming you

Reviewer: ```suggestion
production and non-production instances of Terra, SAM, and TDR, provided you
```

have access. You also use your account for programmatic interactions with the

> above and the Google Cloud resources they own, like the BiqQuery datasets and

Reviewer: ```suggestion
above systems and the Google Cloud resources they host, like the BiqQuery datasets and
```

to dedicated SAM groups (an extension of a Google group). This must be
performed manually by someone with administrator access to that SAM group. For

> non-production instances of TDR, the indexer service account should be added to

Reviewer: ```suggestion
non-production instances of TDR, the indexer service account needs to be added to
```

deployment's service account, for your burner you must follow the steps below:

1. Log into Google Cloud by running

    ```
    gcloud auth login
    ```

    A browser window opens to complete the authentication flow interactively.
    When being prompted, select your burner.

    For more information refer to the Google authorization
    [documentation](https://cloud.google.com/sdk/docs/authorizing).

2. Register your burner with SAM. Run

    ```
    (account="$(gcloud config get-value account)"
    token="$(gcloud auth --account $account print-access-token)"
    curl $AZUL_SAM_SERVICE_URL/register/user/v1  -d "" -H "Authorization: Bearer $token")
    ```

3. Ask an administrator of the `azul-dev` group to add your burner to the
   group. The best way to reach an administrator is via the `#team-boardwalk`
   channel on Slack. Also, ask for a link to the group and note it in your
   records.

4. If you've already attempted to create your deployment via `make deploy`,
   visit the link, sign in as your burner and add your deployment's service
   account to the group. Run `make deploy` again.

For production, use the same procedure, but substitute `azul-dev` with

> `azul-prod` and "burner" with "account".

Reviewer 2: I have a suspicion that this section still applies, just to our personal developer accounts rather than our burner accounts. We still need to authenticate with Terra (and therefore need an account registered with SAM); we just won't be using our burners for this purpose any longer. I think the only way I could be wrong is if developer account registration is already documented somewhere else.
